### PR TITLE
OCPBUGS-11944: use tokenrequest from lib-go

### PR DESCRIPTION
**Pull request by stlaz**
/cc @openshift/openshift-team-auth

### CODE REVIEW

**Comment by stlaz** [comment]
/retest-required
installation failures

File: test/extended/util/oauthserver/oauthserver.go [hunk 1]
```diff
@@ -491,8 +493,8 @@ func determineImageFromRelease(oc *exutil.CLI) (string, error) {
 	return util.DetermineImageFromRelease(oc, "oauth-server")
 }
 
-func newRequestTokenOptions(config *restclient.Config, oauthServerURL, oauthClientName, username, password string) *tokencmd.RequestTokenOptions {
-	options := tokencmd.NewRequestTokenOptions(config, nil, username, password, false)
+func newRequestTokenOptions(config *restclient.Config, oauthServerURL, oauthClientName, username, password string) *tokenrequest.RequestTokenOptions {
+	options := tokenrequest.NewRequestTokenOptions(config, false, challengehandlers.NewBasicChallengeHandler("", os.Stdin, os.Stdout, nil, username, password))
```
Comment (liouk) [inline test/extended/util/oauthserver/oauthserver.go:497]: Why do we now use `""` as the host arg when creating a `BasicChallengeHandler`? The previous implementation would use `config.Host` instead.

Comment (stlaz) [reply test/extended/util/oauthserver/oauthserver.go:497]: I see. It shouldn't be relevant to how it's used but I'll fix it.

**Review by soltysh — APPROVED** [review]
/approve

**Comment by stlaz** [comment]
/test e2e-gcp-ovn-builds
installation failure

**Pull request merged**